### ISSUE TITLE
petrophysical joint inversion of ERT and SRT

### ISSUE THREAD

**Issue by butchi15** [gimli-org/gimli#251]
I am trying to do the petrophysical joint inversion described in section 3.3 of the pyGIMLi paper (https://cg17.pygimli.org) and the code was downloaded from pyGIMLi examples (plot_petro_joint_inv.py) for my field data (ERT and SRT).
ERT data was acquired 48 electrodes with 3m spacing using Dipole-dipole array (Syscal Pro) and SRT data was acquired 24 geophones with 5m spacing using ABEM seismograph. In SRT 9 shots were taken at -30m, -15m, -5m, 27.5m, 57.5m, 87.5m, +5m or120m, +15m or130m and +30m or145m. Both ERT and SRT profiles on the same line but their center points are at same point. 
Now my question is both shot points as well as number of geophones should be same to run this SRT inversion?. or it works in my case also?.   

While trying to import SRT data i am getting an error message "'TravelTimeManager' object is not iterable". or some times its giving as no "t" values in the data set , Zero values in the data set.

runfile('C:/Users/Babu/testing examples/bb_petro_joint_inv.py', wdir='C:/Users/Babu/testing examples')
saving: geometry.pdf ...
.. done
saving: mesh.pdf ...
23/08/20 - 00:18:32 - pyGIMLi - INFO - Found 3 regions.
23/08/20 - 00:18:32 - pyGIMLi - INFO - Region with smallest marker set to background (marker=1)
.. done
Traceback (most recent call last):

  File "C:\Users\Babu\testing examples\bb_petro_joint_inv.py", line 67, in <module>
    ra.invert(ra, mesh=paraMesh)

  File "C:\Users\Babu\anaconda3\envs\pg\lib\site-packages\pygimli\physics\traveltime\TravelTimeManager.py", line 353, in invert
    slowness = super().invert(data, mesh, **kwargs)

  File "C:\Users\Babu\anaconda3\envs\pg\lib\site-packages\pygimli\frameworks\methodManager.py", line 694, in invert
    dataVals = self._ensureData(self.fop.data)

  File "C:\Users\Babu\anaconda3\envs\pg\lib\site-packages\pygimli\frameworks\methodManager.py", line 317, in _ensureData
    if abs(min(vals)) < 1e-12:

TypeError: 'TravelTimeManager' object is not iterable

### Your environment
Operating system:. Windows 10
Python version:3.7.6
pyGIMLi version:"1.1.0+31.g47c8d810"
Way of installation:  Conda package, 
Spider IDE using to run the code: Spyder 4.1.3 

### Steps to reproduce

The following is the code i am using

```python
import pygimli as pg
import numpy as np

import pygimli.meshtools as mt
from pygimli.physics import TravelTimeManager
from pygimli.physics import Refraction
from pygimli.physics.petro import transFwdArchieS as ArchieTrans
from pygimli.physics.petro import transFwdWyllieS as WyllieTrans
from pygimli.frameworks import PetroInversionManager
from pygimli.frameworks import JointPetroInversionManager

def createSynthModel():
    """Return the modelling mesh, the porosity distribution and the
       parametric mesh for inversion.
    """
    # Create the synthetic model
    # create geometry definision for the modelling domain
    world = mt.createWorld (start=[0, 0], end=[141, -40], layers=[-4, -12], worldMarker=False)

    # Create a hetrogenious block
    block = mt.createRectangle (start=[68, -13], end=[76, -20], marker=4, boundaryMarker=10)

    # Merge geometrical entities
    geom = mt.mergePLC([world, block])
    pg.show(geom, boundaryMarker=True, savefig='geometry.pdf')
    
    # Create a mesh from geometry definision
    mesh = mt.createMesh(geom, quality=34)
    pg.show(mesh, savefig='mesh.pdf')
    
    # Create a map to set resistivity values in the appropriate regions
    # [[regionNumber, resistivity], [regionNumber, resistivity], [...]
    rhomap = [[1, 1250.],
              [2, 350.],
              [3, 1500.],
              [4, 10]]

    petro = pg.solver.parseArgToArray(rhomap, mesh.cellCount(), mesh)

    # Create the parametric mesh that only reflect the domain geometry
    world = mt.createWorld (start=[0, 0], end=[141, -40], layers=[-4, -11], worldMarker=False)
    paraMesh = pg.meshtools.createMesh(world, sensors=48, q=34.0)
    ArchieTrans()
    
    return mesh, paraMesh, petro

# Create synthetic model
# ......................
mMesh, paraMesh, saturation = createSynthModel()

# Create Petrophysical models
ertTrans = ArchieTrans(rFluid=20, phi=0.3)
res = ertTrans(saturation)

ttTrans = WyllieTrans(vm=4000, phi=0.3)
vel = 1./ttTrans(saturation)

sensors = paraMesh.positions()[paraMesh.findNodesIdxByMarker(-99)]

 
#%% importing field ERT data
data_ert = pg.load("bbbra.txt", verbose=False)
pg.show(data_ert)

#%% importing field SRT data
ra = Refraction("bbbsa.sgt")
ra.invert(ra, mesh=paraMesh)
ra.showResult(rays=True)  # this works well
   
# %% do ERT inversion
ert = pg.physics.ert.ERTManager(data_ert)
ert.invert(data_ert, mesh=paraMesh)  # maxIter=0 to speed things up


# %% use ERT mesh for SRT inversion
pd = ert.paraDomain
for c in pd.cells():
    c.setMarker(1)
data_srt = Refraction('bbbsa.sgt')
data_srt.setMesh(pd)
data_srt.invert()
data_srt.showResult(rays=True)
```

### Expected behavior

Tell us what should happen or what you want to achieve.

### Actual behavior

Tell us what happens instead and/or provide the output of your script.

```
[bbbsa.txt](https://github.com/gimli-org/gimli/files/5112918/bbbsa.txt)


Paste output here.
[bbbsa.txt](https://github.com/gimli-org/gimli/files/5112918/bbbsa.txt)
[raca_topo.txt](https://github.com/gimli-org/gimli/files/5112912/raca_topo.txt)
[bbbra.txt](https://github.com/gimli-org/gimli/files/5112913/bbbra.txt)




```

> If possible, please add one or more labels to your issue, e.g. if you expect that your issue is rather a question than a problem with the code, please add the label "question".

**Comment by halbmy** [comment]
A few comments:
1. A petrophysical joint inversion for ERT and SRT (surface measurements) as in the example will of course only work if the all the Archie and Wyllie parameters are homogeneous and known (well one could invert for them as well). After my experience this is only rarely the case but of course you can try.
2. If you want to invert for a saturation model, your synthetic model should be a saturation distribution as well, and not a self-set resistivity for which you don't know whether it can be described by an Archie model. The same holds for the seismic velocity.
3. For importing a format like res2dinv, you need pyBERT, calling
```
import pybert as pb
data_ert = pb.importData("bbra.txt")
data_ert.save("bbra.dat", "a b m n rhoa")  #for 4.
```
4. Your SRT data format is wrong. In the header all sensors (shots and geophones) should be present and in the data section, only indices are used.
5. When creating the mesh, you should include all ERT and SRT sensors as nodes.

**Comment by butchi15** [comment]
Thank you for the quick response

**Comment by butchi15** [comment]
Thank you i have corrected my SRT data format, now its reading the file.
importing ERT data using your suggested code through pybert, showing error as "No module named as pybert"


  File "C:\Users\Babu\testing examples\bb_petro_joint_inv.py", line 62, in <module>
    import pybert as pb

ModuleNotFoundError: No module named 'pybert'.

kindly suggest me where i am doing mistake?

**Comment by halbmy** [comment]
The pyBERT module can be installed via conda by
```
conda install -c conda-forge -c gimli pybert
```
See also BERT gitlab project page: https://gitlab.com/resistivity-net/bert

Closed in favor of #253
